src/main/java/org/snomed/snowstorm/core/data/services/RefsetDescriptorUpdaterService.java

```diff
@@ -84,19 +83,46 @@ public void preCommitCompletion(Commit commit) throws IllegalStateException {
 		String branchPath = commit.getBranch().getPath();
 		for (SearchHit<QueryConcept> searchHit : searchHits.getSearchHits()) {
 			long conceptIdL = searchHit.getContent().getConceptIdL();
-			if (Concepts.REFSET.equals(String.valueOf(conceptIdL))) {
+			String conceptIdS = String.valueOf(conceptIdL);
+			if (Concepts.REFSET.equals(conceptIdS)) {
 				// Edge case where first importing.
 				continue;
 			}
 
-			doAddToRefset(commit, conceptIdL, branchPath);
+			List<ReferenceSetMember> members = referenceSetMemberService.findMembers(Concepts.REFSET_DESCRIPTOR_REFSET, branchPath, conceptIdS);
+			if (!members.isEmpty()) {
+				doUpdateRefsetMembers(commit, conceptIdS, branchPath, members);
+			} else {
+				doAddMemberToRefset(commit, conceptIdL, branchPath);
+			}
 		}
 	}
 
-	private void doAddToRefset(Commit commit, long conceptIdL, String branchPath) {
+	private void doUpdateRefsetMembers(Commit commit, String conceptIdS, String branchPath, List<ReferenceSetMember> members) {
+		logger.info("Commit is updating a new Reference Set (conceptId: {}). {} |Reference set descriptor| will be updated accordingly on branch {}.", conceptIdS, Concepts.REFSET_DESCRIPTOR_REFSET, branchPath);
+
+		// Inactivate or remove previous
+		for (ReferenceSetMember member : members) {
+			if (member.isReleased()) {
+				member.setActive(false);
+			} else {
+				member.markDeleted();
+			}
+			member.markChanged();
```

Comment:
I believe I have done the right thing here (setting inactive and marking unreleased as deleted) but please let me know if I have not.